src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -334,6 +380,40 @@ private Object readResolve() {
     }
 
     /**
+     * Record time taken during a certain class of operation in this build.
+     * @param kind what sort of operation is being done
+     * @param end false to start, true at the end (pair in a {@code finally}-block)
+     */
+    synchronized void time(@Nonnull TimingKind kind, boolean end) {
+        if (timings == null) {
+            timings = new HashMap<>();
+        }
+        if (starts == null) {
+            starts = new HashMap<>();
+        }
+        if (end) {
+            Long orig = timings.get(kind.name());
+            if (orig == null) {
+                orig = 0L;
```

Comment:
Huh? `orig == null` if this is the first timing of its kind we have performed in this build.